src/swap_amm.py

```diff
@@ -324,25 +325,21 @@ def __init__(self, symbol1, symbol2, addresses_dict, myswap_id):
     async def get_balance(self):
         if self.myswap_id is not None:
             myswap_pool = await src.blockchain_call.get_myswap_pool(self.myswap_id)
-        for amm, addresses in self.addresses_dict.items():
+        for amm, address in self.addresses_dict.items():
             for token in self.tokens:
-                balance = 0
-                if isinstance(addresses, str):
-                    addresses = [addresses]
-                for address in addresses:
-                    balance += await src.blockchain_call.balance_of(token.address, address)
-                if self.myswap_id is not None:
+                balance = await src.blockchain_call.balance_of(token.address, address)
+                if self.myswap_id is not None and amm == "MySwap":
                     balance += myswap_pool[token.symbol.upper()]
+                balance = decimal.Decimal(balance) / token.decimal_factor
                 self.balances[amm][token.symbol] = balance
-                token.balance_base = sum(self.balances[amm][token.symbol] for amm in self.balances)
-                token.balance_converted = decimal.Decimal(token.balance_base) / token.decimal_factor
+                token.balance_base = balance
+                token.balance_converted = balance
```

Comment:
Remove one of the field for balance then. And name the remaining one just `balance`.